pipeline/load.py

```diff
@@ -31,42 +48,150 @@ def get_data_from_db(conn: connection, table: str, column = "*")-> pd.DataFrame:
-        execute_values(cur, """INSERT INTO author (article_url, source, created_at,
-                       author) VALUES %s""", tuples)
-        cur.commit()
+        tuples = df.to_records(index=False)
+        execute_values(cur, """INSERT INTO article (article_url, source, created_at)
+                       VALUES %s;""", tuples)
+        conn.commit()
+
+
+def add_to_article_author_table(conn: connection, df: pd.DataFrame) -> None:
+    """Converts df into tuples, then adds to author_article table.
+    NB: needs author and article columns converted into foreign key reference"""
 
+    with conn.cursor() as cur:
+        tuples = df.to_records(index=False)
+        execute_values(cur, """INSERT INTO article_author (article_id, author_id)
+                       VALUES %s;""", tuples)
+        conn.commit()
 
-def add_to_scraping_info_table_from_pandas(conn: connection, df: pd.DataFrame) -> None:
+def add_to_scraping_info_table(conn: connection, df: pd.DataFrame) -> None:
     """Converts df into tuples, then adds to scraping_info table.
     NB: needs article_id column converted into foreign key reference"""
 
     with conn.cursor() as cur:
-        tuples = df.itertuples() # is this a list?
+        tuples = df.to_records(index=False)
         execute_values(cur, """INSERT INTO scraping_info (scraped_at, title, body,
                        article_id) VALUES %s""", tuples)
-        cur.commit()
+        conn.commit()
 
 
-if __name__ == "__main__":
-    conn = get_db_connection()
-    df = pd.read_csv(TRANSFORMED_DATA)
+def retrieve_author_id(conn: connection, name: str) -> str:
+    """Retrieves author_id from author table."""
+
+    with conn.cursor() as cur:
+        cur.execute("""SELECT author_id from author where author_name = %s""", [name])
+        author_id = cur.fetchall()
+        if len(author_id) == 1:
+            return author_id[0][0]
+    return None
 
-    df_no_duplicates = check_for_duplicates(conn, df, "article", "article_url")
-    conn.close()
+
+def retrieve_article_id(conn: connection, url: str) -> str:
+    """Retrieves article_id from author table."""
+
+    with conn.cursor() as cur:
+        cur.execute("""SELECT article_id from article where article_url = %s""", [url])
+        url = cur.fetchall()
+        if len(url) == 1:
+            return url[0][0]
+    return None
+
+
+def add_to_author_table(conn: connection, df: pd.DataFrame) -> None:
+    """Converts df into tuples, then adds to authors table.
+    NB: needs article_id column converted into foreign key reference"""
+
+    with conn.cursor() as cur:
+        tuples = df.to_records(index=False)
+        execute_values(cur, """INSERT INTO author (author_name) VALUES %s;""", tuples)
+        conn.commit()
+
+
+def add_to_article_version_table(conn: connection, df: pd.DataFrame) -> None:
+    """Converts df into tuples, then adds to article_version table.
+    NB: needs article_id column converted into foreign key reference"""
```

Comment:
is this note still needed?

Reply:
Do you mean the docstring? Because the article_id is a foreign key so it's needed as a disclaimer

Reply:
ah okay I thought it was like a TODO later type thing, nvm